numpyro/distributions/continuous.py

```diff
@@ -2533,3 +2533,133 @@ def variance(self):
             theoretical_var *= 1 - 1 / self.event_shape[axis]
 
         return jnp.broadcast_to(theoretical_var, self.batch_shape + self.event_shape)
+
+
+class Wishart(Distribution):
+    """
+    Wishart distribution for covariance matrices.
+
+    :param concentration: Positive concentration parameter analogous to the
+        concentration of a :class:`Gamma` distribution. The concentration must be larger
+        than the dimensionality of the scale matrix.
+    :param scale_matrix: Scale matrix analogous to the inverse rate of a :class:`Gamma`
+        distribution.
+    :param rate_matrix: Rate matrix anaologous to the rate of a :class:`Gamma`
+        distribution.
+    :param scale_tril: Cholesky decomposition of the :code:`scale_matrix`.
+    """
+
+    arg_constraints = {
+        "concentration": constraints.dependent(is_discrete=False),
+        "scale_matrix": constraints.positive_definite,
+        "rate_matrix": constraints.positive_definite,
+        "scale_tril": constraints.lower_cholesky,
+    }
+    support = constraints.positive_definite
+    reparametrized_params = [
+        "scale_matrix",
+        "rate_matrix",
+        "scale_tril",
+    ]
+
+    def __init__(
+        self,
+        concentration=None,
+        scale_matrix=None,
+        rate_matrix=None,
+        scale_tril=None,
+        validate_args=None,
+    ):
+        # Determine the shapes.
+        batch_shape = None
+        event_shape = None
+        for x in [scale_matrix, rate_matrix, scale_tril]:
+            if x is not None:
+                batch_shape = jnp.broadcast_shapes(
+                    jnp.shape(concentration), jnp.shape(x)[:-2]
+                )
+                event_shape = jnp.shape(x)[-2:]
+                break
+        if event_shape is None:
+            raise ValueError(
+                "One of `scale_matrix`, `rate_matrix`, or `scale_tril` must be "
+                "specified."
+            )
+
+        # Coerce to scale_tril parameter.
+        if scale_matrix is not None:
+            self.scale_matrix = jnp.broadcast_to(
+                scale_matrix, batch_shape + event_shape
+            )
+            self.scale_tril = jnp.linalg.cholesky(scale_matrix)
```

Comment:
We need to promote shapes like in MultivariateNormal distribution. No need to broadcast. This saves computation when `concentration` is batched.